fix(code size): revert previous devMode change to restore size targets

yjbanov opened the pull request:
This commit reverts https://github.com/angular/angular/commit/a8d9dbf110496075617b956654894cf3e7f0d190 that introduced a code size regression (16kb gzipped, 63kb minified) in Dart.

Effect on the hello world app:

gzipped: 105kb -> 89kb
minified: 370kb -> 317kb

BREAKING CHANGE:
- This is very unlikely to be breaking, but I'm still marking just in case. The only change to the user should be that dev mode is driven by Dart's checked mode, like it was in the past.

Closes #5883
Closes #5888

yjbanov commented:
/cc @vsavkin @IgorMinar

IgorMinar commented on gulpfile.js:
```diff
@@ -638,7 +638,7 @@ gulp.task('test.unit.dart', function(done) {
 // This test will fail if the size of our hello_world app goes beyond one of
 // these values when compressed at the specified level.
 // Measure in bytes.
-var _DART_PAYLOAD_SIZE_LIMITS = {'uncompressed': 375 * 1024, 'gzip level=6': 105 * 1024};
```
please don't change limits in a drive-by way. in the future please create separate commit and explain why you are decreasing the limit.

yjbanov replied:
Done

IgorMinar commented:
I believe that I saw some tests in g3 that required dev mode even when not running in dartium's checked mode.

can you please verify how to deal with this (by cs-ing enableDevMode) before we merge this in?

yjbanov committed:
fix(code size): revert previous devMode change to restore size targets

This commit reverts a8d9dbf that introduced a code size regression (16kb gzipped, 63kb minified) in Dart.

Effect on the hello world app:

gzipped: 105kb -> 89kb
minified: 370kb -> 317kb

BREAKING CHANGE:
- This is very unlikely to be breaking, but I'm still marking just in case. The only change to the user should be that dev mode is driven by Dart's checked mode, like it was in the past.

yjbanov committed:
chore(build): tighten up code size check now that regression is fixed

Our code size SLA is 100kb gzipped and 300kb minified. Our target is 10kb gzipped.

We are currently under 90kb gzipped. Let's keep it that way while looking for ways to improve the situation further. We're not <300kb minified yet, so we should be stricter here too.

IgorMinar commented:
this PR was still failing on CI even after rebasing on top for @vsavkin's PR.

@vsavkin landed 3dca9d522a79d037b3c2deba537ed547fc3f65b8 in the meantime, can you please rebase?

mary-poppins commented:
Merging PR #5896 on behalf of @alexeagle to branch [presubmit-alexeagle-pr-5896](https://github.com/angular/angular/tree/presubmit-alexeagle-pr-5896).
